ts-client/src/amm/index.ts

```diff
@@ -0,0 +1,782 @@
+  }
+
+  public async swap(owner: PublicKey, inTokenMint: PublicKey, inAmountLamport: BN, outAmountLamport: BN) {
+    const [sourceToken, destinationToken] =
+      this.tokenA.address === inTokenMint.toBase58()
+        ? [this.poolState.tokenAMint, this.poolState.tokenBMint]
+        : [this.poolState.tokenBMint, this.poolState.tokenAMint];
+
+    const adminTokenFee =
+      this.tokenA.address === inTokenMint.toBase58() ? this.poolState.adminTokenAFee : this.poolState.adminTokenBFee;
+
+    let preInstructions: Array<TransactionInstruction> = [];
+    const [[userSourceToken, createUserSourceIx], [userDestinationToken, createUserDestinationIx]] =
+      await this.createATAPreInstructions(owner, [sourceToken, destinationToken]);
+
+    createUserSourceIx && preInstructions.push(createUserSourceIx);
+    createUserDestinationIx && preInstructions.push(createUserDestinationIx);
+
+    if (sourceToken.equals(WRAPPED_SOL_MINT)) {
+      preInstructions = preInstructions.concat(wrapSOLInstruction(owner, userSourceToken, inAmountLamport.toNumber()));
+    }
+
+    preInstructions.push(await this.getApySyncInstructions());
+
+    const postInstructions: Array<TransactionInstruction> = [];
+    if (WRAPPED_SOL_MINT.equals(destinationToken)) {
+      const unwrapSOLIx = await unwrapSOLInstruction(owner);
+      unwrapSOLIx && postInstructions.push(unwrapSOLIx);
+    }
+
+    const swapTx = await this.program.methods
+      .swap(inAmountLamport, outAmountLamport)
+      .accounts({
+        aTokenVault: this.vaultA.vaultState.tokenVault,
+        bTokenVault: this.vaultB.vaultState.tokenVault,
+        aVault: this.poolState.aVault,
+        bVault: this.poolState.bVault,
+        aVaultLp: this.poolState.aVaultLp,
+        bVaultLp: this.poolState.bVaultLp,
+        aVaultLpMint: this.vaultA.vaultState.lpMint,
+        bVaultLpMint: this.vaultB.vaultState.lpMint,
+        userSourceToken,
+        userDestinationToken,
+        user: owner,
+        adminTokenFee,
+        pool: this.address,
+        tokenProgram: TOKEN_PROGRAM_ID,
+        vaultProgram: this.vaultProgram.programId,
+      })
+      .remainingAccounts(getRemainingAccounts(this.poolState))
+      .preInstructions(preInstructions)
+      .postInstructions(postInstructions)
+      .transaction();
+
+    return new Transaction({
+      feePayer: owner,
+      ...(await this.program.provider.connection.getLatestBlockhash('finalized')),
+    }).add(swapTx);
+  }
+
+  public async getDepositQuote(
+    tokenAInAmount: BN,
+    tokenBInAmount: BN,
+    slippage?: BN,
+  ): Promise<{
+    poolTokenAmountOut: BN;
+    tokenAInAmount: BN;
+    tokenBInAmount: BN;
+  }> {
+    const slippageRate = slippage ?? DEFAULT_SLIPPAGE;
+    const { tokenAAmount, tokenBAmount } = await this.getPoolInfo();
+
+    const { poolLpSupply, vaultALpSupply, vaultBLpSupply, poolVaultALp, poolVaultBLp } = await this.getSplInfo();
+
+    if (tokenAInAmount.eq(new BN(0))) {
```

Comment:
How stable pool get quote for depositing only single token ? Eg: deposit only token B. 
`if (tokenAInAmount.eq(new BN(0)))` and `if (tokenBInAmount.eq(new BN(0)))` blocked single deposit for stable pool

Reply:
added 2 test case (single deposit for stable pool)